### PR TITLE
README: Update project info and technical aims/support

### PR DESCRIPTION
**Pull request by valeriabarra**
Given the fact that I will be presenting ClimaCore at AGU next week, we might have new visitors to our GH page :) Hence, I would like to provide some introductory information in our README. 
Here is a mix of what @simonbyrne had produced for PASC and what I adapted/added for AGU.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
 
 [zenodo-img]: https://zenodo.org/badge/356355994.svg
 [zenodo-url]: https://zenodo.org/badge/latestdoi/356355994
+
+The Climate Modelling Alliance ([CliMA](https://clima.caltech.edu/)) project is developing a full Earth System Model (ESM), entirely in the high-level, dynamic [Julia](https://julialang.org/) language. The main goal of the project is to build an ESM that automatically learns from diverse data sources to produce accurate climate predictions with quantified uncertainties. The CliMA model has been developed to target both CPU and GPU architectures, using a common codebase. ClimaCore.jl constitutes the dynamical core (_dycore_), providing discretization tools to solve the governing equations describing the different components of the ESM.
```
Comment (tapios) [inline README.md:34]:
```suggestion
The Climate Modelling Alliance ([CliMA](https://clima.caltech.edu/)) is developing a new Earth System Model (ESM), entirely written in the [Julia](https://julialang.org/) language. The main goal of the project is to build an ESM that automatically learns from diverse data sources to produce accurate climate predictions with quantified uncertainties. The CliMA model targets both CPU and GPU architectures, using a common codebase. ClimaCore.jl constitutes the dynamical core (_dycore_) of the atmosphere and land models, providing discretization tools to solve the governing equations of the ESM component models.
```

File: README.md [hunk 2]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+ClimaCore.jl's high-level application programming interface (API) facilitates modularity and composition of differential operators and the definition of flexible discretizations. This, in turn, is coupled with the low-level API that supports different data layouts, specialized implementations and flexible models for threading, to better face high-performance optimization, data storage, and scalability challenges on modern HPC ecosystems.
```
Comment (tapios) [inline README.md:35]:
```suggestion
ClimaCore.jl's high-level application programming interface (API) facilitates modularity and composition of differential operators and the definition of flexible discretizations. This, in turn, is coupled with low-level APIs that support different data layouts, specialized implementations, and flexible models for threading, to better face high-performance optimization, data storage, and scalability challenges on modern HPC architectures.
```

File: README.md [hunk 3]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+
+## Technical aims and current support
+* Support both Large Eddy Simulation (LES) and General Circulation Model (GCM) configurations.
+* Limit number of languages (ideally one: currently, Julia).
```
Comment (tapios) [inline README.md:39]:
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+ClimaCore.jl's high-level application programming interface (API) facilitates modularity and composition of differential operators and the definition of flexible discretizations. This, in turn, is coupled with the low-level API that supports different data layouts, specialized implementations and flexible models for threading, to better face high-performance optimization, data storage, and scalability challenges on modern HPC ecosystems.
+
+## Technical aims and current support
+* Support both Large Eddy Simulation (LES) and General Circulation Model (GCM) configurations.
```
Comment (tapios) [inline README.md:38]:
```suggestion
* Support both large-eddy simulation (LES) and general circulation model (GCM) configurations for the atmosphere.
```

File: README.md [hunk 5]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Limit number of languages (ideally one: currently, Julia).
+* A suite of tools for constructing spatial compatible discretizations.
```
Comment (tapios) [inline README.md:40]:
```suggestion
* A suite of tools for constructing space discretizations.
```

File: README.md [hunk 6]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Horizontal spectral elements:
+    - Supports both Continuous (SEM) and Discontinuous Galerkin (DG-SEM) discretizations.
```
Comment (tapios) [inline README.md:42]:
```suggestion
    - Supports both continuous Galerkin (CG) and discontinuous Galerkin (DG) spectral element discretizations.
```

File: README.md [hunk 7]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Flexible choice of vertical discretization (currently staggered Finite Differences, potentially Finite Volumes)
```
Comment (tapios) [inline README.md:43]:
```suggestion
* Flexible choice of vertical discretization (currently staggered finite differences)
```

File: README.md [hunk 8]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Support for different geometric vector types and coordinate systems: Covariant and Contravariant vectors for curvilinear, non-orthogonal systems; Cartesian vectors for Eucledean spaces; and UVVectors for lat-long grids.
```
Comment (tapios) [inline README.md:44]:
```suggestion
* Support for different geometries (Cartesian, spherical), with governing equations discretized in terms of covariant  vectors for curvilinear, non-orthogonal systems and Cartesian vectors for Euclidean spaces.
```

File: README.md [hunk 9]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* `Field` abstraction:
+    - Scalar, vector or struct-valued
+    - Stores values, geometry, and mesh information
+    - Flexible memory layouts: Array-of-Structs (AoS), Struct-of-Arrays (SoA),Array-of-Struct-of-Arrays (AoSoA)
+    - Useful overloads: `sum` (integral), `norm`, etc.
+    - Compatible with [`DifferentialEquations.jl`](https://diffeq.sciml.ai/stable/) time steppers.
+* Composable operators via broadcasting: Apply a function element-wise to an array; Scalar values are broadcast over arrays Fusion of multiple operations; Can be specialized for custom functions or argument types (e.g. `CuArray` compiles and applies a custom CUDA kernel).
```
Comment (tapios) [inline README.md:51]:
```suggestion
* Composable operators via broadcasting: apply a function element-wise to an array; scalar values are broadcast over arrays 
* Fusion of multiple operations; can be specialized for custom functions or argument types (e.g. `CuArray` compiles and applies a custom CUDA kernel).
```

File: README.md [hunk 10]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Operators (`grad`, `div`, `interpolate`) are “pseudo-functions”: Act like functions when broadcasted over a `Field`; Fuse operators and function calls.
```
Comment (tapios) [inline README.md:52]:
```suggestion
* Operators (`grad`, `div`, `interpolate`) are “pseudo-functions”: Act like functions when broadcasted over a `Field`; fuse operators and function calls.
```

File: README.md [hunk 11]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+* Add element node sizes dimensions to type domain
```
Comment (tapios) [inline README.md:53]:
```suggestion
* Add element node size dimensions to type domain
```

File: README.md [hunk 12]
```diff
@@ -32,3 +30,29 @@ CliMA model dycore
...
+    - i.e. specialize on polynomial degree
```
Comment (tapios) [inline README.md:54]:
```suggestion
    - i.e., specialize on polynomial degree
```

File: README.md (hunk 11 above)
Comment (jakebolewski) [reply README.md:53]: I think this is mostly an implementation detail

**Comment by jakebolewski** [comment]
Would it be better for this to go in the documentation?  I feel that a couple sentences explaining what the Dycore is useful to give an overview, but Ideally the README addresses this is what it is, this is how I install it, this is how I do something useful (quickstart).

File: README.md (hunk 12 above)
Comment (jakebolewski) [reply README.md:54]: also kind of an implementation detail, not sure if it is worth highlighting

**Comment by valeriabarra** [comment]
> Would it be better for this to go in the documentation? I feel that a couple sentences explaining what the Dycore is useful to give an overview, but Ideally the README addresses what the project is about, how I install it, and how I do something useful (quickstart) while pointing to more info.

I just thought that since (hopefully) we will have some new visitors to our GH landing page, it was a bit of a shame to present an empty README that only shows badges as of now. I meant to give a broader intro/overview, but yes, we could also think of adding more sections on 'How to' or 'Getting started' type of guides. And we could include the content from the README in the docs?

**Comment by jakebolewski** [comment]
A small gif of one of the paraview movies would be nice as well

File: README.md (hunk 11 above)
Comment (valeriabarra) [reply README.md]: Ok, we can remove it from here

Comment (valeriabarra) [reply README.md]: but I think its nested bullet points are useful and I thought the sentence served to give context

**Comment by valeriabarra** [comment]
> A small gif of one of the paraview movies would be nice as well

Yeah... I have this one so far. What do you think? Too coarse? This was a pretty coarse run, but I am also trying to produce fine(r)-resolution ones in the meantime. 
barotropic_instability_vorticity_day10_with_bg_short

**Comment by valeriabarra** [comment]
PS: unrelated to this, but how should I have added the `fieldname` so that it doesn't show as "data"? Should I have passed a `fieldname = ...` or `name = ...` option to `writevtk` ?

**Comment by jakebolewski** [comment]
Add support for it: https://github.com/CliMA/ClimaCore.jl/blob/main/lib/ClimaCoreVTK/src/ClimaCoreVTK.jl#L55

you can change it also in paraview

**Comment by valeriabarra** [comment]
> Add support for it: https://github.com/CliMA/ClimaCore.jl/blob/main/lib/ClimaCoreVTK/src/ClimaCoreVTK.jl#L55 https://github.com/CliMA/ClimaCore.jl/blob/main/lib/ClimaCoreVTK/src/addfield.jl#L43
> 
> you can change it also in paraview

I don't think we need to modify the `vtk_file` function. I saw that we support named tuples already, but only for the single snapshot `writevtk` version. I attempted a fix for the time series one in #345 but needs to be revisited, as it doesn't work yet.

**Comment by valeriabarra** [comment]
@simonbyrne do you have any comments/opinions on this?

**Review by simonbyrne — APPROVED** [review]
Thanks @valeriabarra: this looks good to me!

**Pull request merged**